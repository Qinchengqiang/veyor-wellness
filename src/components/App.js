import '../assets/styles/App.scss';
import {shallowEqual, useDispatch, useSelector} from "react-redux";
import {doLogin} from "../actions/loginAction";

// import loadable from "@loadable/component";

function App() {
    const authRedux = useSelector(state => state.auth, shallowEqual)
    const dispatch = useDispatch();

    const login = (
        <>
            {authRedux.isAuthenticated ?
                <h4>{authRedux.user.username}</h4>
                :
                <p>
                    <button onClick={() => dispatch(doLogin())}>login</button>
                </p>
            }
        </>);

    return (
        <div className="App container d-flex flex-column align-items-center" data-testid={`App`}>

            <h3>SPA</h3>

        </div>
    );
}

export default App;
