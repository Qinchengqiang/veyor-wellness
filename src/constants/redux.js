export const SET_USER_REQUEST = 'SET_USER_REQUEST';     // for saga watch func
export const SET_CURRENT_USER = 'SET_CURRENT_USER';

