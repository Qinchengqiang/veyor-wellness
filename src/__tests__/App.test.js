import {cleanup, render, screen} from '@testing-library/react';  // testing-library
import renderer from 'react-test-renderer';
import React from 'react';
import App from '../components/App';
import {applyMiddleware, createStore} from "redux";
import rootReducer from "../reducers";
import {composeWithDevTools} from "redux-devtools-extension";
import logger from "redux-logger";
import {Provider} from "react-redux";
import {BrowserRouter as Router, Route} from "react-router-dom";
import createSagaMiddleware from 'redux-saga';
import rootSage from '../sagas';


/** must wrap the app component since the redux */
const AppWrapper = () => {

    /** saga */
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(
        rootReducer,
        composeWithDevTools(
            applyMiddleware(sagaMiddleware, logger)
        )
    );
    sagaMiddleware.run(rootSage);

    const routes = (
        <>
            <Route exact path="/" component={App}/>
        </>);


    // if (localStorage.jwtToken) {
    //     setAxiosAuth(user);
    //     store.dispatch(setCurrentUser(user))    // keep login status after re-loading page
    // }

    return (
        <Provider store={store}>
            <Router>
                {routes}
            </Router>
        </Provider>
    )
}

afterEach(() => {
    cleanup();
});

describe('App group:', () => {
    // test 1
    it('App test 1: render app', () => {
        render(<AppWrapper/>);
        const AppEle = screen.getByTestId('App');   // attr in element: data-testid={`App`}
        expect(AppEle).toBeInTheDocument();
    })

    // test 2
    test('App test 2: SPA', () => {
        render(<AppWrapper/>);
        const linkElement = screen.getByText(/SPA/i);
        expect(linkElement).toBeInTheDocument();
    });
})


/**
 * snapshot test (using react-test-renderer)
 */
test('App test: matches snapshot', () => {
    const tree = renderer.create(<AppWrapper/>).toJSON();
    expect(tree).toMatchSnapshot();                 // failed，press ‘u’ to update. (jest -u)
})

