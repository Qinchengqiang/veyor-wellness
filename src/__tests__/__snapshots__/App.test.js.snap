// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test: matches snapshot 1`] = `
<div
  className="App container d-flex flex-column align-items-center"
  data-testid="App"
>
  <h3>
    SPA
  </h3>
</div>
`;
